src/ORM/Behavior/TreeBehavior.php

```diff
@@ -797,15 +797,20 @@ protected function _getPrimaryKey()
 /**
  * Returns the depth level of a node in the tree.
  *
- * @param int|string $id Primary key of the node.
+ * @param int|string|\Cake\Datasource\EntityInterface $entity The entity or primary key get the level of.
  * @return int|bool Integer of the level or false if the node does not exist.
  */
-    public function getLevel($id)
+    public function getLevel($entity)
     {
+        $primaryKey = $this->_getPrimaryKey();
+        $id = $entity;
+        if (!is_scalar($entity)) {
```

Comment:
ya or that but something to check if the `$entity` can actually do the `get()` and throw an exception otherwise (php would in any case but I think it should be handled by the framework).